lib/resources/shim-head.js

```diff
@@ -2,6 +2,25 @@
 ((win) => {
   win.ELECTRON = true;
 
+  if (process.env.ELECTRON_PROTOCOL_SERVE_INDEX && window.location.protocol !== 'file:') {
```

Comment:
Wouldn't it be better to use something like `=== 'serve:'` here? What if the window navigates to an `http://` URL, for example? I don't think we'd want to add this back in then since I thought the original, upstream code only applied to `file://`.